Close quarters mimery

heepox opened the pull request:
## About The Pull Request
My first PR please be gentle and give me some feedback

Mime traitor martial art...
Might need to tweak some values and such no sure if its too powerful or too weak.

SURPRISE_SPIN_COMBO "DH" Makes the dude do the spin emote and makes them confused for a time (still gotta tweak confusion time i think)

TOUNGE_PULL_COMBO "GHG" Tounge damage and silence for a bit (cannot be performed if the target does not have a tounge)

THROAT_PUNCH_COMBO "HHG" like krav maga lung punch but uhh weaker and without the CLAMP causing no breath just regular breathloss instead.

ARM_PULL_COMBO "GD" like carp wrist wrench but weaker stun and no shoving stunbaton man away (does not fit maybe delete)

MIME_SPECIAL_COMBO "DGGHHD" Rip out tounge destroying it and lots of bleeding and damage and a stun so its a 6 long combo (not sure about qdeling tounge) (cannot be performed if the target does not have a tounge)

Currently 9TC just so you can have it and advanced mimery.
## Why It's Good For The Game

Just one more mime traitor item making the mime have a stunning TWO unique traitor items

## Changelog
:cl:
add: Close quarters Mimery so mimes have something more than advanced mimery and potato
/:cl:

heepox committed:
Update beestation.dme

mimery

heepox committed:
Update melee.dm

heepox committed:
Create close_quarters_mimery.dm

heepox committed:
Update granters.dm

heepox committed:
Update uplink_items.dm

heepox committed:
Update close_quarters_mimery.dm

forgot to change cqm to close_quarters_mimery
fixed that

heepox committed:
Update uplink_items.dm

deleted an extra space above cqm woops

Archanial commented:
Martial arts are awful and combos are bad, we shouldn't have more of them to be honest.

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+#define TOUNGE_PULL_COMBO "GHG" //
+#define THROAT_PUNCH_COMBO "HHG" //
+#define ARM_PULL_COMBO "GD" //
+#define MIME_SPECIAL_COMBO "DGHGHD" //
```
```suggestion
#define SURPRISE_SPIN_COMBO "DD"
#define TOUNGE_PULL_COMBO "GHG"
#define THROAT_PUNCH_COMBO "HHG"
#define ARM_PULL_COMBO "GD"
#define MIME_SPECIAL_COMBO "DGHGHD"
```

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+	if(findtext(streak,SURPRISE_SPIN_COMBO))
+		streak = ""
+		headTwist(A,D)
+		return 1
```
```suggestion
		return TRUE
```

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+	if(findtext(streak,TOUNGE_PULL_COMBO))
+		streak = ""
+		toungePull(A,D)
+		return 1
```
```suggestion
		return TRUE
```

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+	if(findtext(streak,THROAT_PUNCH_COMBO))
+		streak = ""
+		throatPunch(A,D)
+		return 1
```
```suggestion
		return TRUE
```

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+	if(findtext(streak,ARM_PULL_COMBO))
+		streak = ""
+		armPull(A,D)
+		return 1
```
```suggestion
		return TRUE
```

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+	if(findtext(streak,MIME_SPECIAL_COMBO))
+		streak = ""
+		mimeSpecial(A,D)
+		return 1
```
```suggestion
		return TRUE
```

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+		streak = ""
+		mimeSpecial(A,D)
+		return 1
+	return 0
```
```suggestion
	return FALSE
```

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+	D.confused += 20
+	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+	playsound(get_turf(D), 'sound/weapons/thudswoosh.ogg', 30, 1, -1)
+	return 1
```
```suggestion
	if(!D.stat && !D.IsStun() && !D.IsParalyzed())
		log_combat(A, D, "Surprise spun (Close Quarters Mimery)")
		D.visible_message("<span class='warning'>[A] spins [D] right round like a record!</span>", \
		"<span class='userdanger'>[A] spins you right round like a record!</span>")
		D.emote("spin")
		D.confused += 20
		A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
		playsound(get_turf(D), 'sound/weapons/thudswoosh.ogg', 30, 1, -1)
		return TRUE
```
It doesn't compile.
You didn't even test it.

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+	playsound(get_turf(D), 'sound/weapons/blobattack.ogg', 30, 1, -1)
+	if(D.silent <= 10)
+        	D.silent = CLAMP(D.silent + 10, 0, 10)
+	return 1
```
```suggestion
	return TRUE
```

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+	playsound(get_turf(D), 'sound/weapons/blobattack.ogg', 30, 1, -1)
+	if(D.silent <= 10)
+        	D.silent = CLAMP(D.silent + 10, 0, 10)
```
```suggestion
	if(D.silent <= 10)
    	D.silent = CLAMP(D.silent + 10, 0, 10)
```

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+	D.losebreath += 3
+	D.adjustOxyLoss(10)
+	if(D.silent <= 15)
+        	D.silent = CLAMP(D.silent + 15, 0, 15)
```
same as above, use tabs not spaces....

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+	D.adjustOxyLoss(10)
+	if(D.silent <= 15)
+        	D.silent = CLAMP(D.silent + 15, 0, 15)
+	return 1
```
```suggestion
	return TRUE
```

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+	D.Stun(30)
+	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+	playsound(get_turf(A), 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
+	return 1
```
That won't compile
see above

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+	if(D.getorgan(/obj/item/organ/tounge))
+		tounge.Remove(D)
+		qdel(tounge)
+	return 1
```
```suggestion
	return TRUE
```

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+	var/obj/item/organ/tounge/tounge = D.getorganslot(ORGAN_SLOT_TOUNGE)
+	if(D.getorgan(/obj/item/organ/tounge))
+		tounge.Remove(D)
+		qdel(tounge)
```
```suggestion
	var/obj/item/organ/tounge/T = D.getorganslot(ORGAN_SLOT_TOUNGE)
	if(T)
		T.Remove(D)
		qdel(T)
```

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+/datum/martial_art/cqm/harm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	add_to_streak("H",D)
+	if(check_streak(A,D))
+		return 1
```
```suggestion
		return TRUE
```

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+/datum/martial_art/cqm/grab_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	add_to_streak("G",D)
+	if(check_streak(A,D))
+		return 1
```
```suggestion
		return TRUE
```

Archanial commented on code/datums/martial/close_quarters_mimery.dm:
```diff
@@ -0,0 +1,136 @@
+/datum/martial_art/cqm/disarm_act(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	add_to_streak("D",D)
+	if(check_streak(A,D))
+		return 1
```
```suggestion
		return TRUE
```

Archanial commented on code/game/objects/items/granters.dm:
```diff
@@ -359,6 +359,33 @@
+
+/obj/item/book/granter/martial/close_quarters_mimery/already_known(mob/user)
+	if(usr && usr.mind)
+		if(!usr.mind.miming)
```
```suggestion
	if(user.mind && !user.mind.miming)
```

Archanial reviewed (changes requested):
Martial arts could use a rewrite anyway

heepox committed:
Update code/datums/martial/close_quarters_mimery.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update code/datums/martial/close_quarters_mimery.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update code/datums/martial/close_quarters_mimery.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update code/datums/martial/close_quarters_mimery.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update code/datums/martial/close_quarters_mimery.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update code/datums/martial/close_quarters_mimery.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update code/datums/martial/close_quarters_mimery.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update code/datums/martial/close_quarters_mimery.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update code/datums/martial/close_quarters_mimery.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update code/game/objects/items/granters.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update code/datums/martial/close_quarters_mimery.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update code/datums/martial/close_quarters_mimery.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update code/datums/martial/close_quarters_mimery.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update code/datums/martial/close_quarters_mimery.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update code/datums/martial/close_quarters_mimery.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update code/datums/martial/close_quarters_mimery.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update code/datums/martial/close_quarters_mimery.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update close_quarters_mimery.dm

fixes some spaces out.

heepox committed:
Update close_quarters_mimery.dm

TRUE instead of return 1

Crossedfall commented:
Based on the commits, it looks like you might be trying to do all of this via the web editor. I'd strongly advise looking at the following guide on setting up a proper development environment: https://forums.beestation13.com/t/github-building-the-hive/1334

I'd also recommend getting VSCode to properly edit and test the files.

heepox commented:
> Martial arts are awful and combos are bad, we shouldn't have more of them to be honest.

if this does not pan out ill propably make an equivalent item to the clowns tae-clown-do shoes for the mime.

heepox committed:
Update beestation.dme

heepox committed:
Update close_quarters_mimery.dm

heepox committed:
Rename close_quarters_mimery.dm to cqm.dm

heepox committed:
Update granters.dm

heepox committed:
Update uplink_items.dm

heepox commented:
so i fixed some spelling errors and removed a few tabs and spaces. maybe a few more adjustments and this might be a moment where someone said to me WYCI and i did.

heepox commented:
currently just testing it out... might need to adjust combos so they dont conflict

heepox committed:
Update cqm.dm

adjusted combo to avoid conflict

heepox commented:
no more combo conflicts, but mime special is a really nasty attack. gonna lessen the bloodloss unless its fine as a very long combo death dealer. hmm

heepox committed:
Update cqm.dm

prevents stacking confusion to absurd levels

heepox committed:
Update cqm.dm

less bleeding so they dont INSTANTLY DIE TO SUFFOCATION!

super12pl commented:
9 TC for martial art that lets you (practically) permanently silence someone is well.... questionable

Archanial commented on code/datums/martial/cqm.dm:
```diff
@@ -0,0 +1,140 @@
+		A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
+		playsound(get_turf(D), 'sound/weapons/thudswoosh.ogg', 30, 1, -1)
+		return TRUE
+
```
```suggestion
```
random newline

Archanial commented on code/datums/martial/cqm.dm:
```diff
@@ -0,0 +1,140 @@
+	D.apply_damage(10, A.dna.species.attack_type, BODY_ZONE_HEAD, def_check)
+	D.adjustOrganLoss(ORGAN_SLOT_TONGUE, 15, 200)
+	D.Jitter(20)
+	A.do_attack_animation(D, ATTACK_EFFECT_PUNCH)
```
this should have a check to see if target has tongue

Archanial commented on code/datums/martial/cqm.dm:
```diff
@@ -0,0 +1,140 @@
+
+//Mime special, RIP HIS TOUNGE OUT AND CRUSH IT BEFORE HIS EYES! 25 brute to the head and a 6s stun also destroying the targets tounge and making them BLEED!
+/datum/martial_art/cqm/proc/mimeSpecial(mob/living/carbon/human/A, mob/living/carbon/human/D)
+	var/obj/item/organ/tongue/T = D.getorganslot(ORGAN_SLOT_TONGUE)
```
if(T) check should be way above. Whole move doesn't make sense if a person doesn't have tongue in first place.

Archanial commented on code/game/objects/items/granters.dm:
```diff
@@ -359,6 +359,32 @@
+		to_chat(user, "<span class='warning'>You try to read the scroll but can't comprehend any of it.</span>")
+		return TRUE
+	else
+		return FALSE
```
This logic is pretty weird. You are returning true if a player does not have mind or is not a mime or else your return false?

heepox replied:
yeah i just need !user.mind.miming so user.mind is not needed. return true means cannot learn in granters false means can learn i know weird.

heepox replied:
and yes i tested it... only mimes that have a vow of silence can read it, nobody else

heepox committed:
Update code/datums/martial/cqm.dm

Co-authored-by: Archanial <37456678+Archanial@users.noreply.github.com>

heepox committed:
Update granters.dm

removing spaghetti

heepox committed:
Update cqm.dm

checking for tounges on tounge related moves.

heepox commented:
> 9 TC for martial art that lets you (practically) permanently silence someone is well.... questionable

losing your tounge does not prevent you from speaking just makes it harder to understand what you are saying

like Help mime bad can turn into h'lp mim' ba'

it basically either replaces a letter or two with aa oo or ' but otherwise leaves the words intact.

also an active radio jammer in your backpack is more effective at silencing people than this martial art and jammer is 3tc

heepox committed:
Update cqm.dm

more fitting attack effect for spin

Goblinu commented:
> 
> 
> > Martial arts are awful and combos are bad, we shouldn't have more of them to be honest.
> 
> if this does not pan out ill propably make an equivalent item to the clowns tae-clown-do shoes for the mime.

If you do make the shoes invisible

heepox commented:
> > > Martial arts are awful and combos are bad, we shouldn't have more of them to be honest.
> > 
> > 
> > if this does not pan out ill propably make an equivalent item to the clowns tae-clown-do shoes for the mime.
> 
> If you do make the shoes invisible

haha no sprite shoes... eh no i would propably just make them normal shoes or give you a choice list of their appearance to choose from (invisible being one) and then making that the permament sprite for that instance of the item. maybe a mimefu mask instead i dunno.

heepox committed:
Update cqm.dm

adjusting combos

heepox commented:
so everything should work as intended now "should"

ike709 commented:
🔑

ike709 commented:
🔑  by fran
